brave_domains/service_domains.h

```diff
@@ -22,9 +27,14 @@ namespace brave_domains {
 // All domains can be overridden globally via
 // CLI param syntax is brave-services-env={dev,staging,prod}.
 //
-// Prefix overridde(s) take precedence over global override.
+// Precedence is:
+// 1. Prefix specific CLI overrides
+// 2. Global CLI overrides overrides
+// 3. Default env override parameter
+// 4. Default env (production)
 std::string GetServicesDomain(
-    std::string prefix = "",
+    std::string prefix,
+    std::string env_value_default_override = "",
```

Comment:
I wonder if this should be an enum. As it's a string, it's not obvious to the caller what the value they should put in. They might type it out themselves. We could put an enum here (e.g. `ServicesEnvironment::Dev|Staging|Production`) or at least have a `DCHECK` if it's not one of the expected values. ...but that doesn't stop people putting the string in directly...

Reply:
Also I wonder if this should default to dev for everything? I guess I don't understand what the purpose of dev vs staging is - there's possibilities dev instances have in-progress code that isn't suitable for master? But if that's the case then ai chat will suffer...?